app/src/main/java/com/dscvit/handly/ui/home/CollectionsFragment.kt

```diff
@@ -0,0 +1,227 @@
+                dialogView.modify_progress.setIndeterminateDrawable(Circle())
+                dialogView.modify_progress.hide()
+                dialogView.modify_name.setText(collectionAdapter.collectionList[position].name)
+
+                dialogView.modify_button.setOnClickListener {
+                    if (dialogView.modify_name.text.isNotBlank()) {
+                        val updateCollection = UpdateCollection(
+                            collectionAdapter.collectionList[position].id,
+                            dialogView.modify_name.text.toString().trim()
+                        )
+                        homeViewModel.updateCollections(updateCollection)
+                            .observe(viewLifecycleOwner, Observer {
+                                when (it.status) {
+                                    Result.Status.LOADING -> {
+                                        dialogView.modify_name.hide()
+                                        dialogView.modify_cancel.hide()
+                                        dialogView.modify_button.hide()
+                                        dialogView.modify_delete.hide()
+                                        dialogView.modify_title.hide()
+                                        dialogView.modify_progress.show()
+                                    }
+                                    Result.Status.SUCCESS -> {
+                                        getCollection(homeViewModel, collectionAdapter)
+                                        dialogBuilder.dismiss()
+                                    }
+                                    Result.Status.ERROR -> {
+                                        Log.d("esh", it.message!!)
+                                        shortToast("Oops something went wrong")
+
+                                        dialogView.modify_name.show()
+                                        dialogView.modify_cancel.show()
+                                        dialogView.modify_button.show()
+                                        dialogView.modify_delete.show()
+                                        dialogView.modify_title.show()
+                                        dialogView.modify_progress.hide()
+                                    }
+                                }
+                            })
+                    } else {
+                        shortToast("Name can't be empty")
+                    }
+                }
+
+                dialogView.modify_delete.setOnClickListener {
+                    val deleteCollectionRequest = DeleteCollectionRequest(
+                        collectionAdapter.collectionList[position].id
+                    )
+                    homeViewModel.deleteCollection(deleteCollectionRequest)
+                        .observe(viewLifecycleOwner, Observer {
+                            when (it) {
+                                "Loading" -> {
```

Comment:
Consider using constants for this, both for the observer and the livedata emitter.

Reply:
I wont need this later, as rn the backend wasn't sending any response body, later i can just use the normal Result model